dev-tools/mage/integtest.go

```diff
@@ -150,6 +150,11 @@ func RunIntegTest(mageTarget string, test func() error, passThroughEnvVars ...st
 		"TEST_TAGS",
 		"PYTHON_EXE",
 		"MODULE",
+
+		// XXX: Move these variables to some other place?
```

Comment:
this could also be a credentials file under `~/.aws`

Reply:
Yep. With this comment I was thinking more in moving the pass through list of variables to the magefiles of the beats that use them, or use the new helpers Andrew [added](https://github.com/elastic/beats/pull/17773/files#diff-50588e69952adcfd9465814c86fbe56eR809) to add variables using their preffix.
If we pass the credentials using `~/.aws/credentials` we will also need to change the tests, that currently expect the credentials to be in environment variables. We would also need to find something equivalent for Azure and GCP.

Reply:
@kaiyan-sheng @narph would it be possible to use a file for credentials in both Azure and GCP?

I would expect this to be safer having the way we pass env vars, right @jsoriano ?

Reply:
> I would expect this to be safer having the way we pass env vars, right @jsoriano ?

I'd say so, but we have mechanisms to mask the values of sensitive environment variables, and this is something we are already doing in other places.
With files we will have to be sure that they are not in paths exposed in the workspace. But I guess we would be fine if we use any path out of the workspace and we use the `AWS_CONFIG_FILE` to indicate where it is.

Reply:
I didn't know we had this case already, if this is a solved problem let's not introduce something else that is not really needed :+1:

Reply:
Well, not in Beats, but `withEnvMask` is used in other observability projects for these kind of secrets.